compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -5155,7 +5155,12 @@ private BLangLiteral createSimpleLiteral(Node literal, SyntaxKind sign, boolean
                 type == SyntaxKind.TEMPLATE_STRING || type == SyntaxKind.IDENTIFIER_TOKEN) {
             String text = textValue;
             if (type == SyntaxKind.STRING_LITERAL) {
-                text = text.substring(1, text.length() - 1);
+                if (text.length() > 1 && text.charAt(text.length() - 1) == '"') {
```

Comment:
can there be a `text` with length == 1,
please add a test case for something like `string name = """`

Reply:
Yes, an empty string with a missing end quote will be a length of 1. Will add the suggested test case

Reply:
so we don't need first condition right `text.length() > 1 ` in case length is one or less l:5162 will throw a index out of bound expection.

Reply:
for length == 1, `text.substring(1)` will be an empty text. No IndexOutofBoudException ryt?
We cannot have a `STRING_LITERAL` with a length of 0.

We need `text.length() > 1` right? Otherwise for legnth 1(starting double quote only)
if condtion will be true and `text = text.substring(1, text.length() - 1);` will result in IndexOutofBoudException.if length is

Reply:
oh my bad.